integration_tests/test_rust_plugin.py

```diff
@@ -29,6 +29,14 @@ def test_stage_rust_plugin(self):
             os.path.join('stage', 'bin', 'simple-rust'), cwd=project_dir)
         self.assertEqual('There is rust on snaps!\n', binary_output)
 
+    def test_stage_rust_with_revision(self):
+        project_dir = 'simple-rust-with-revision'
+        self.run_snapcraft('stage', project_dir)
+
+        binary_output = self.get_output_ignoring_non_zero_exit(
+            os.path.join('stage', 'bin', 'simple-rust'), cwd=project_dir)
+        self.assertEqual('There is rust on snaps!\n', binary_output)
```

Comment:
This is not checking that we compiled with the right rust.
I think this would be better: http://paste.ubuntu.com/23758746/